contrib/dracut/90zfs/zfs-lib.sh.in

```diff
@@ -128,10 +128,8 @@ export_all() {
     ret=0
 
     IFS="${NEWLINE}"
-    for pool in $(zpool list -H -o name) ; do
-        if zpool list -H "${pool}" > /dev/null 2>&1; then
-            zpool export "${pool}" "$@" || ret=$?
-        fi
+    for pool in $(zpool list -Ho name); do
+        zpool export "${pool}" "$@" || ret=$?
     done
```

Comment:
Is there a reason we can't use `zpool export -a` and get rid of the loop entirely?

```
     -a      Exports all pools imported on the system.
```

Reply:
Historically, yes: 07a3312f170ac56cb480b0df9fdf4c83f116b59b, which originated this, landed on Tue Oct 7 10:20:56 2014 -0700, at which point `zpool export` didn't have `-a`.

Now, though, no. I've replaced the two users with straight `zpool export -aF`.